llpc/patch/llpcPatchIntrinsicSimplify.cpp

```diff
@@ -0,0 +1,413 @@
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcPatchIntrinsicSimplify.cpp
+ * @brief LLPC source file: contains implementation of class Llpc::PatchIntrinsicSimplify.
+ ***********************************************************************************************************************
+ */
+#define DEBUG_TYPE "llpc-patch-intrinsic-simplify"
+
+#if defined(_MSC_VER) || defined(__MINGW32__)
+// According to Microsoft, one must set _USE_MATH_DEFINES in order to get M_PI
+// from the Visual C++ cmath / math.h headers:
+// https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=vs-2019
+#define _USE_MATH_DEFINES
+#endif
+
+#include <math.h>
+
+#include "llvm/InitializePasses.h"
+#include "llvm/Analysis/ScalarEvolution.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
+
+#include "llpcPatchIntrinsicSimplify.h"
+#include "llpcPipelineState.h"
+#include "llpcTargetInfo.h"
+
+using namespace llvm;
+using namespace Llpc;
+
+namespace Llpc
+{
+
+// =====================================================================================================================
+// Initializes static members.
+char PatchIntrinsicSimplify::ID = 0;
+
+// =====================================================================================================================
+// Pass creator, creates the LLVM pass intrinsic simplifcations.
+FunctionPass* CreatePatchIntrinsicSimplify()
+{
+    return new PatchIntrinsicSimplify();
+}
+
+// =====================================================================================================================
+// Constructor.
+PatchIntrinsicSimplify::PatchIntrinsicSimplify()
+    :
+    FunctionPass(ID)
+{
+}
+
+// =====================================================================================================================
+// Get the analysis usage.
+void PatchIntrinsicSimplify::getAnalysisUsage(
+    AnalysisUsage& analysisUsage // [out] The analysis usage for this pass.
+    ) const
+{
+    analysisUsage.addRequired<ScalarEvolutionWrapperPass>();
+    analysisUsage.addPreserved<ScalarEvolutionWrapperPass>();
+    analysisUsage.addRequired<PipelineStateWrapper>();
+    analysisUsage.setPreservesCFG();
+}
+
+// =====================================================================================================================
+// Executes this LLVM patching pass on the specified LLVM module.
+bool PatchIntrinsicSimplify::runOnFunction(
+    Function& func) // [in,out] LLVM module to be run on.
+{
+    bool changed = false;
+    Module* module = func.getParent();
```

Comment:
consider `m_pModule = ...` without the local variable

Reply:
Thanks, that was just dumb.